crates/starknet-types-rpc/src/generated/v0_5_0.rs

```diff
@@ -0,0 +1,1476 @@
+    pub block_id: BlockId,
+    /// The hash of the requested contract class
+    pub class_hash: Felt,
+}
+
+/// Parameters of the `starknet_getClassHashAt` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct GetClassHashAtParams {
+    /// The hash of the requested block, or number (height) of the requested block, or a block tag
+    pub block_id: BlockId,
+    /// The address of the contract whose class hash will be returned
+    pub contract_address: Address,
+}
+
+/// Parameters of the `starknet_getClassAt` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct GetClassAtParams {
+    /// The hash of the requested block, or number (height) of the requested block, or a block tag
+    pub block_id: BlockId,
+    /// The address of the contract whose class definition will be returned
+    pub contract_address: Address,
+}
+
+/// Parameters of the `starknet_getBlockTransactionCount` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct GetBlockTransactionCountParams {
+    /// The hash of the requested block, or number (height) of the requested block, or a block tag
+    pub block_id: BlockId,
+}
+
+/// Parameters of the `starknet_call` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CallParams {
+    /// The details of the function call
+    pub request: FunctionCall,
+    /// The hash of the requested block, or number (height) of the requested block, or a block tag, for the block referencing the state or call the transaction on.
+    pub block_id: BlockId,
+}
+
+/// Parameters of the `starknet_estimateFee` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct EstimateFeeParams {
+    /// The transaction to estimate
+    pub request: Vec<BroadcastedTxn>,
+    /// The hash of the requested block, or number (height) of the requested block, or a block tag, for the block referencing the state or call the transaction on.
+    pub block_id: BlockId,
+}
+
+/// Parameters of the `starknet_estimateMessageFee` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct EstimateMessageFeeParams {
+    /// the message's parameters
+    pub message: MsgFromL1,
+    /// The hash of the requested block, or number (height) of the requested block, or a block tag, for the block referencing the state or call the transaction on.
+    pub block_id: BlockId,
+}
+
+/// Parameters of the `starknet_blockNumber` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BlockNumberParams {}
+
+/// Parameters of the `starknet_blockHashAndNumber` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BlockHashAndNumberParams {}
+
+/// Parameters of the `starknet_chainId` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ChainIdParams {}
+
+/// Parameters of the `starknet_syncing` method.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SyncingParams {}
```

Comment:
And those? Is it just that this RPC route does not take any params?

Reply:
Yep exactly